test/es6/newProxy.js

```diff
@@ -0,0 +1,20 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+let calledHandler = false;
+var func0 = function () {
+};
+var func1 = function () {
+  var v0 = new Proxy(func0, proxyHandler);
+  new v0().a;
+};
+var proxyHandler = {get: function () {
+    calledHandler = true;
+}};
+func1()
+func1()
+func1()
+
+print(calledHandler ? "Fail" : "Pass");

```

Comment:
Chrome and Firefox disagree with this: they call the proxy to fetch "protoype" while creating v0.

Reply:
Ah interesting! I guess I assumed incorrectly that the interpreter was correct/JIT was wrong. I'll dig a bit deeper.

Reply:
I think not calling the proxy might be correct, but honestly I can't really tell:

From [the spec](https://www.ecma-international.org/ecma-262/8.0/index.html#sec-proxy-object-internal-methods-and-internal-slots-construct-argumentslist-newtarget)

The [[Construct]] internal method of a Proxy exotic object O is called with parameters argumentsList which is a possibly empty List of ECMAScript language values and newTarget. The following steps are taken:

1. Let handler be O.[[ProxyHandler]].
2. If handler is null, throw a TypeError exception.
3. Assert: Type(handler) is Object.
4. Let target be O.[[ProxyTarget]].
5. Let trap be ? GetMethod(handler, "construct").
6. If trap is undefined, then
    a. Assert: target has a [[Construct]] internal method.
    b. Return ? Construct(target, argumentsList, newTarget).

Reply:
Agreed, I don't see anything there or in [new](https://tc39.github.io/ecma262/#sec-evaluatenew) or [Construct](https://tc39.github.io/ecma262/#sec-construct) or [IsConstructor](https://tc39.github.io/ecma262/#sec-isconstructor) suggesting that we should fetch "prototype" from the proxy. Not sure why other engines are doing this.

Reply:
No, I think the other engines might be right. The `newTarget` in that part of the spec will be the proxy itself if you are calling `new thingThatIsProxy()`, and then the `Construct(Target, argumentsList, newTarget)` ends up going into `GetPrototypeFromConstructor` for normal function objects, which according to https://www.ecma-international.org/ecma-262/8.0/index.html#sec-getprototypefromconstructor ends up calling `Get(constructor, "prototype")` where `constructor` is the `newTarget`

Reply:
Assuming I read the spec at https://www.ecma-international.org/ecma-262/8.0/index.html#sec-ecmascript-function-objects-construct-argumentslist-newtarget correctly, that is.

Reply:
Then we should try with what happens at Reflect.construct(v0,[])

Reply:
@MSLaguana that seems like it.
//cc @zenparsing

Reply:
That does make a lot of sense. I think CallAsConstructor is nulling out newTarget for proxies when it shouldn't.